src/Test.FunctionalTests.BinSkim.Driver/BaselineTests.cs

```diff
@@ -36,6 +46,125 @@ public void Driver_BuiltInRuleFunctionalTests()
             this.BatchRuleRules(string.Empty, "*.dll", "*.exe", "gcc.*", "clang.*", "macho.*");
         }
 
+        [Fact]
+        public void Driver_ShouldLogCompilerTelemetryEvents_Managed()
+        {
+            List<ITelemetry> sendItems = CompilerTelemetryTestSetup();
+            var sb = new StringBuilder();
+            string testDirectory = PEBinaryTests.BaselineTestDataDirectory + Path.DirectorySeparatorChar;
+            string testFile = Path.Combine(testDirectory, "DotNetCore_win-x86_VS2019_Default.dll");
+
+            SarifLog sarifResult = RunRules(sb, testFile);
+
+            string testEnviornmentVar = Environment.GetEnvironmentVariable(nameof(AnalysisSummaryExtractor.ProjectIdVariableName), EnvironmentVariableTarget.Process);
+
+            sendItems.All<ITelemetry>(item => item.GetType() == typeof(EventTelemetry));
+            List<EventTelemetry> events = sendItems.OfType<EventTelemetry>().ToList();
+
+            List<EventTelemetry> compilerEvents = events.Where(e => e.Name == CompilerDataLogger.CompilerEventName)
+                .OfType<EventTelemetry>().ToList();
+
+            List<EventTelemetry> assemblyReferencesEvents = events.Where(e => e.Name == CompilerDataLogger.AssemblyReferencesEventName)
+                .OfType<EventTelemetry>().ToList();
+
+            List<EventTelemetry> commandLineEvents = events.Where(e => e.Name == CompilerDataLogger.CommandLineEventName)
+                .OfType<EventTelemetry>().ToList();
+
+            List<EventTelemetry> summaryEvents = events.Where(e => e.Name == CompilerDataLogger.SummaryEventName)
+                .OfType<EventTelemetry>().ToList();
+
+            if (summaryEvents.Count != 1)
+            {
+                sb.AppendLine("Expected at least 1 summaryEvent, but found none.");
+            }
+            if (assemblyReferencesEvents.Count != 1)
+            {
+                sb.AppendLine(string.Format("Expected 1 AssemblyReferencesEvent, but found {0}", assemblyReferencesEvents.Count));
+            }
+            if (commandLineEvents.Count != 0)
+            {
+                sb.AppendLine(string.Format("Expected 0 CommandLineEvents, but found {0}", commandLineEvents.Count));
+            }
+            if (compilerEvents.Count != 1)
+            {
+                sb.AppendLine(string.Format("Expected 1 CompilerEvent, but found {0}", compilerEvents.Count));
+            }
+
+            string summaryEventSessionId = summaryEvents.First().Properties["sessionId"];
```

Comment:
> summaryEvents [](http://example.com/codeflow?start=42&length=14)

add something like:

summaryEvents.Should().NotBeNull();

so that there's a good failfast test msg if this invariant breaks.